packages/google_maps_flutter/google_maps_flutter/lib/src/google_map.dart

```diff
@@ -282,6 +284,12 @@ class GoogleMap extends StatefulWidget {
   /// were not claimed by any other gesture recognizer.
   final Set<Factory<OneSequenceGestureRecognizer>> gestureRecognizers;
 
+  /// Identifier that's associated with a specific cloud-based map style.
+  ///
+  /// See https://developers.google.com/maps/documentation/get-map-id
+  /// for more details.
+  final String? cloudMapId;
```

Comment:
Alternately we could leave that to do at the application level, with some example code; having platform-based logic isn't really much different from populating a platform structure. E.g.,

```dart
String? _getMapId() {
  if (kIsWeb) {
    return someWebKey;
  } else if (Platform.isIOS) {
    return someIOSKey;
  } else if (Platform.isAndroid) {
    return someAndroidKey;
  }
  return nil;
}
```

A multi-platform config object is problematic because it prevents 3P implementations for new platforms.

Reply:
This approach sounds good to me! I think we can make this part of an example app or similar.